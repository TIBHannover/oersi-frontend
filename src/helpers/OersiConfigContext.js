import React from "react"

const OersiConfigContext = React.createContext(undefined)

export default OersiConfigContext
