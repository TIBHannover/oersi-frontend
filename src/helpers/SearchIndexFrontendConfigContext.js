import React from "react"

const SearchIndexFrontendConfigContext = React.createContext(undefined)

export default SearchIndexFrontendConfigContext
