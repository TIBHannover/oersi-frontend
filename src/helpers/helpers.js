import {i18n} from "next-i18next"
import moment from "moment"
import "moment/locale/de"

/**
 * function to get the location and return a value for  specific query parameters
 * @param {Location} location Get location
 * @param {string} queryToSearch String to check if exist or not in URL example: queryToSearch="pageSize"
 */
export function getParams(router, queryToSearch) {
  const searchParams = new URLSearchParams(router.query)
  if (searchParams.has(queryToSearch) === true)
    return searchParams.get(queryToSearch)
  else return null
}

/**
 *
 * @param {Location} location Get location
 * @param {Object} queryToInsertUpdate Contain key/value for setting or Ubdating Params in URL example {name:'size',value:5}
 */
export function setParams(location, queryToInsertUpdate) {
  const addUpdateParams = new URLSearchParams(location.search)
  addUpdateParams.set(queryToInsertUpdate.name, queryToInsertUpdate.value)
  return addUpdateParams
}

export function getThumbnailUrl(resourceId) {
  const fileId =
    resourceId && resourceId.length > 250 ? resourceId.substring(0, 250) : resourceId
  return process.env.NEXT_PUBLIC_PUBLIC_URL + "/thumbnail/" + fileId + ".webp"
}

export function getDisplayValue(rawValue, fieldOption, translateFnc) {
  if (rawValue === "N/A") {
    return translateFnc("LABEL.N/A")
  } else if (fieldOption?.defaultDisplayType === "licenseGroup") {
    return getLicenseGroupById(rawValue).toUpperCase()
  }
  return processFieldOption(rawValue, fieldOption, translateFnc)
}
export function processFieldOption(value, fieldOption, translateFnc) {
  let result = value
  if (fieldOption?.translationNamespace) {
    const translate = (v) =>
      translateFnc(fieldOption.translationNamespace + "#" + v, {
        keySeparator: false,
        nsSeparator: "#",
      })
    result = Array.isArray(value) ? value.map(translate) : translate(value)
  }
  return result
}

/**
 * Get the group for the given license
 * @param {string} licenseId
 */
export function getLicenseGroupById(licenseId) {
  if (licenseId) {
    if (
      licenseId
        .toLowerCase()
        .startsWith("https://creativecommons.org/publicdomain/mark")
    ) {
      return "PDM"
    } else if (licenseId.match("^https?://www.apache.org/licenses/.*")) {
      return "Apache"
    } else if (
      licenseId.match("^https?://(www.)?opensource.org/licenses?/0?[bB][sS][dD].*")
    ) {
      return "BSD"
    } else if (licenseId.match("^https?://www.gnu.org/licenses/[al]?gpl.*")) {
      return "GPL"
    } else if (licenseId.match("^https?://www.gnu.org/licenses/fdl.*")) {
      return "FDL"
    } else if (
      licenseId.match("^https?://(www.)?opensource.org/licenses?/[mM][iI][tT]")
    ) {
      return "MIT"
    }
    const regex =
      /^https?:\/\/[a-zA-Z0-9.-]+\/(?:licenses|licences|publicdomain)(?:\/publicdomain)?\/([a-zA-Z-]+)/g
    let match = regex.exec(licenseId)
    if (match) {
      return match[1]
    }
  }
  return ""
}
export function getLicenseLabel(license) {
  let regex =
    /^https?:\/\/creativecommons.org\/(?:licenses|licences|publicdomain)(?:\/publicdomain)?\/([a-zA-Z-]+)(?:\/([0-9.]+))?(?:\/([a-z]{2})(?:$|\/))?/g
  let match = regex.exec(license)
  if (match) {
    let label
    const group = match[1].toLowerCase()
    const version = match[2]
    const country = match[3]
    if (group === "mark") {
      label = "Public Domain Mark"
    } else if (group === "zero") {
      label = "CC0"
    } else {
      label = "CC " + group.toUpperCase()
    }
    if (version) {
      label += " " + version
    }
    if (country) {
      label += " " + country.toUpperCase()
    }
    return label
  }
  regex = /^https?:\/\/www.apache.org\/licenses\/LICENSE-([0-9.]+)/g
  match = regex.exec(license)
  if (match) {
    const version = match[1]
    return "Apache " + version
  }
  regex = /^https?:\/\/(?:www.)?opensource.org\/licenses?\/[mM][iI][tT]/g
  match = regex.exec(license)
  if (match) {
    return "MIT"
  }
  regex = /^https?:\/\/(?:www.)?opensource.org\/licenses?\/(0?[bB][sS][dD].*)/g
  match = regex.exec(license)
  if (match) {
    return match[1]
  }
  regex = /^https?:\/\/www.gnu.org\/licenses\/([al]?gpl)(?:-([0-9.]+))?/g
  match = regex.exec(license)
  if (match) {
    let label = "GNU " + match[1].toUpperCase()
    const version = match[2]
    if (version) {
      label += " " + version
    }
    return label
  }

  return ""
}

/**
 * function that check if a string is valid Url or not
 * @param {string} str an Url as string to check if is valid or not
 * @returns {boolean} value, true if is valid
 */
export function isValidURL(str) {
  var pattern = new RegExp("(www.|http://|https://|ftp://)")
  return pattern.test(str)
}

/**
 * function that build a url with a path
 * @param {string} str an path to attach in Url
 * @returns {string} return complete url
 */
export function buildUrl(str) {
  var urlBuild = process.env.NEXT_PUBLIC_PUBLIC_URL
  if (str) {
    urlBuild = urlBuild + "/" + str
  }
  return new URL(urlBuild)
}

/**
 * Function that determines the privacy-policy-link from the given links matches the given language-code (or fallback-lng)
 * @param {Array} privacyPolicyLinks All link from Configuration
 * @param {String} lang  Language Code from Translate
 */
export function getPrivacyPolicyLinkForLanguage(
  privacyPolicyLinks,
  lang,
  fallBackLang
) {
  let policyEntry = undefined
  if (privacyPolicyLinks || privacyPolicyLinks instanceof Array) {
    policyEntry = Array.from(privacyPolicyLinks).filter(
      (item) => item["language"] === lang && item["path"]
    )[0]
    if (policyEntry === undefined) {
      policyEntry = Array.from(privacyPolicyLinks).filter(
        (item) => fallBackLang.includes(item["language"]) && item["path"]
      )[0]
    }
  }

  if (policyEntry !== undefined)
    return !isValidURL(policyEntry["path"])
      ? buildUrl(policyEntry["path"])
      : policyEntry["path"]

  return undefined
}

export function formatDate(date, format) {
  if (date !== null) {
    moment.locale(i18n.language)
    return moment(date).format(format)
  } else {
    return ""
  }
}

export function getSafeUrl(url) {
  const whitelistProtocols = ["http", "https"]
  return whitelistProtocols.find((p) => url?.toString().startsWith(p + ":"))
    ? url
    : ""
}

export function getEmbedValues(embeddingConfig, baseFieldValues, record) {
  return {
    ...baseFieldValues,
    mediaUrl: getSafeUrl(getValueFromRecord(embeddingConfig?.mediaUrl, record)),
    fallbackMediaUrl: embeddingConfig?.fallbackMediaUrl
      ?.map((e) =>
        getValuesFromRecord({field: e}, record).map((e) => getSafeUrl(e.field))
      )
      .flat(),
  }
}
export function getBaseFieldValues(baseFieldConfig, record) {
  const getRawValues = (fieldName) =>
    getValuesFromRecord({field: fieldName}, record)
      .filter((e) => e.field)
      .map((e) => e.field)
  const licenseUrl = getSafeUrl(
    getValueFromRecord(baseFieldConfig.licenseUrl, record)
  )
  return {
    title: getValueFromRecord(baseFieldConfig.title, record),
    resourceLink: getSafeUrl(
      getValueFromRecord(baseFieldConfig.resourceLink, record)
    ),
    description: getValueFromRecord(baseFieldConfig.description, record),
    keywords: getRawValues(baseFieldConfig.keywords),
    author: getRawValues(baseFieldConfig.author),
    licenseUrl: licenseUrl,
    licenseGroup: getLicenseGroupById(licenseUrl).toLowerCase(),
    thumbnailUrl: getSafeUrl(
      getValueFromRecord(baseFieldConfig.thumbnailUrl, record)
    ),
  }
}
export function getValueFromRecord(fieldName, record) {
  if (!fieldName || !record) {
    return null
  }
  const values = getValuesFromRecord({field: fieldName}, record)
  return values.length > 0 ? values[0].field : null
}
/**
 *
 * @param fields get values for these fields; format is a map fieldId -> fieldName, example {field: "name", linkField: "url"}
 * @param record the record (or sub-record) to get the values from
 * @returns a list containing all found values, each entry of the form {<fieldId1>: <value1>, <fieldId2>: <value2>}
 */
export function getValuesFromRecord(fields, record) {
  if (!record) {
    return []
  }
  const hasSubfields = (f) => f.includes(".")
  const getCurrentFieldName = (f) => f.substring(0, f.indexOf("."))
  const getNextSubfieldName = (f) => f.substring(f.indexOf(".") + 1)
  const getFieldValue = (fieldName, object) => {
    if (object?.hasOwnProperty(fieldName)) {
      return object[fieldName]
    }
    return ""
  }
  const fieldsByCurrentFieldName = Object.entries(fields)
    .filter(([k, v]) => v)
    .map(([k, v]) => {
      if (hasSubfields(v)) {
        return {cur: getCurrentFieldName(v), next: getNextSubfieldName(v), id: k}
      }
      return {cur: v, next: null, id: k}
    })
    .reduce((groups, item) => {
      const group = groups[item.cur] || []
      group.push(item)
      groups[item.cur] = group
      return groups
    }, {})
  const combine = (a, b) => a.map((e1) => b.map((e2) => ({...e1, ...e2}))).flat()
  let resultList = []
  for (let [curField, items] of Object.entries(fieldsByCurrentFieldName)) {
    const value = getFieldValue(curField, record)
    const values = Array.isArray(value) ? value : [value]
    const curFieldResultList = values
      .map((v) => {
        const result = {}
        items
          .filter((e) => !e.next)
          .forEach((e) => {
            result[e.id] = v
          })
        const nextItems = items
          .filter((e) => e.next)
          .reduce((fields, item) => {
            fields[item.id] = item.next
            return fields
          }, {})
        if (Object.keys(nextItems).length > 0) {
          return combine([result], getValuesFromRecord(nextItems, v))
        }
        return [result]
      })
      .flat()
    resultList =
      resultList.length === 0
        ? curFieldResultList
        : combine(resultList, curFieldResultList)
  }
  return resultList
}
