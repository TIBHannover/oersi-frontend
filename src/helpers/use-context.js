import * as React from "react"

export const SearchIndexFrontendConfigContext = React.createContext(undefined)
