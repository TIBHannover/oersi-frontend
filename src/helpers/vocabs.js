/**
 * Add entries for parentIds that are not included in the given data, but that are contained in a used path of the data
 * @param data
 * @param parentIdMap
 * @returns {*}
 */
function addMissingParentIds(data, parentIdMap) {
  const ids = data.map((d) => d.key)
  const addNonPresentParentId = (list, d) => {
    list.push(d)
    const parentIdExists = d.parentId !== null && d.parentId !== undefined
    if (
      parentIdExists &&
      !ids.includes(d.parentId) &&
      !list.map((i) => i.key).includes(d.parentId)
    ) {
      list = [
        {
          key: d.parentId,
          parentId: d.parentId in parentIdMap ? parentIdMap[d.parentId] : null,
          doc_count: 0,
        },
      ].reduce(addNonPresentParentId, list)
    }
    return list
  }
  return data.reduce(addNonPresentParentId, [])
}

/**
 *
 * @param dataList the flat vocab-list that should be converted
 * @param parentIdMap the scheme of the vocab describing the hierarchical structure of the vocab - just the map of id->parentId
 */
export function toHierarchicalList(dataList, parentIdMap) {
  if (!dataList || !parentIdMap) {
    return []
  }
  let extendedList = dataList.map((d) => ({
    ...d,
    parentId: d.key in parentIdMap ? parentIdMap[d.key] : null,
  }))
  extendedList = addMissingParentIds(extendedList, parentIdMap)
  const addChildren = (p) => {
    return {
      ...p,
      children: extendedList.filter((d) => d.parentId === p.key).map(addChildren),
    }
  }
  const convertToVocabNodes = (d) => {
    const node = new HierarchicalVocabNode(d.key, d.doc_count, d)
    if (d.children.length) {
      node.children = d.children.map(convertToVocabNodes).map((e) => {
        e.parent = node
        return e
      })
    }
    return node
  }
  return extendedList
    .filter((d) => d.parentId === null)
    .map(addChildren)
    .map(convertToVocabNodes)
}

export function getSiblings(node) {
  if (node.parent === null) {
    return []
  }
  return node.parent.children.filter((d) => d.key !== node.key)
}

class HierarchicalVocabNode {
  constructor(key, doc_count, originalData = null) {
    this.key = key
    this.doc_count = doc_count
    this.parent = null
    this.children = []
    this.originalData = originalData
  }
}
export class HierarchicalDataPreparer {
  constructor(data, parentIdMap) {
    this.data = toHierarchicalList(data, parentIdMap)
  }
  modifyNodes(modFcn) {
    modifyAll(this.data, modFcn)
    return this
  }
}

export function findAllChildNodes(node, filterFcn) {
  const getNodes = (list, e) => {
    if (filterFcn(e)) {
      list.push(e)
    }
    if (e.children.length > 0) {
      list = e.children.reduce(getNodes, list)
    }
    return list
  }
  return node.children?.length ? node.children.reduce(getNodes, []) : []
}
export function modifyAll(nodes, modFcn) {
  return nodes.map((node) => {
    modFcn(node)
    if (node.children?.length) {
      modifyAll(node.children, modFcn)
    }
    return node
  })
}
export function modifyAllParents(node, modFcn) {
  if (node.parent) {
    modFcn(node.parent)
    modifyAllParents(node.parent, modFcn)
  }
}
