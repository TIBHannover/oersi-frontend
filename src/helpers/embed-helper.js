import {joinArrayField} from "./helpers"

/**
 * Check if an embed-snippet can be generated for the given dataset.
 * @param {Object} data data to check
 */
export function isEmbeddable(data) {
  if (data.licenseGroup) {
    if (
      [
        "by",
        "by-nc",
        "by-nc-nd",
        "by-nc-sa",
        "by-nd",
        "by-sa",
        "pdm",
        "zero",
      ].includes(data.licenseGroup)
    ) {
      return !citationNeedsAuthor(data) || hasAuthor(data)
    }
  }
  return false
}

/**
 * Check if the given dataset has at least one author.
 * @param {Object} data data to check
 */
function hasAuthor(data) {
  if (!data.creator) {
    return false
  }
  return data.creator.length > 0
}

/**
 * Check if the author is required in the citation of the given dataset.
 * @param {Object} data data to check
 */
function citationNeedsAuthor(data) {
  return data.licenseGroup.startsWith("by")
}

/**
 * Get the html embedding code for the given data.
 * @param {Object} data data
 * @param {Object} t translation function
 */
export function getHtmlEmbedding(data, t) {
  const htmlMedia = getHtmlEmbeddingMedia(data, t, defaultMediaMapping)
  const htmlCaption = getHtmlEmbeddingCaption(data, t)
  if (htmlMedia) {
    return `<!-- OERSI: embed ${data.id} -->
<figure class="embedded-material">
    ${htmlMedia}
    <figcaption>
        ${htmlCaption}
    </figcaption>
</figure>
`
  }
  return `<!-- OERSI: embed ${data.id} -->
<div class="embedded-material">
    ${htmlCaption}
</div>
`
}

const defaultMediaMapping = [
  {
    regex: "https://av.tib.eu/media/([0-9]+)",
    html: (match) =>
      `<iframe width="560" height="315" scrolling="no" src="//av.tib.eu/player/${match[1]}" frameborder="0" allowfullscreen></iframe>`,
  },
]

/**
 * Get the html embedding code for the media part.
 * @param {Object} data data
 * @param {Object} t translation function
 * @param {Object} mediaMapping mapping from source url to embedding-code for media
 */
function getHtmlEmbeddingMedia(data, t, mediaMapping) {
  if (mediaMapping) {
    let m
    for (let i = 0; i < mediaMapping.length; i++) {
      m = mediaMapping[i]
      const regex = new RegExp(m.regex)
      const match = regex.exec(data.id)
      if (match) {
        return m.html(match)
      }
    }
  }
  return ""
}
function getHtmlEmbeddingCaption(data, t) {
  let caption = `<q><a href="${data.id}">${data.name}</a></q>`
  if (citationNeedsAuthor(data)) {
    caption +=
      " " +
      t("EMBED_MATERIAL.BY") +
      " " +
      joinArrayField(data.creator, (item) => item.name, null)
  }
  caption += ` ${t("EMBED_MATERIAL.UNDER")} <a href="${
    data.license
  }">${getLicenseLabel(data.license)}</a>`
  return caption
}

export function getLicenseLabel(license) {
  const regex = /^https?:\/\/creativecommons.org\/(?:licenses|licences|publicdomain)(?:\/publicdomain)?\/([a-zA-Z-]+)(?:\/([0-9.]+))?(?:\/([a-z]+))?/g
  let match = regex.exec(license)
  if (match) {
    let label
    const group = match[1].toLowerCase()
    const version = match[2]
    const country = match[3]
    if (group === "mark") {
      label = "Public Domain Mark"
    } else if (group === "zero") {
      label = "CC0"
    } else {
      label = "CC " + group.toUpperCase()
    }
    if (version) {
      label += " " + version
    }
    if (country) {
      label += " " + country.toUpperCase()
    }
    return label
  }
  return ""
}
