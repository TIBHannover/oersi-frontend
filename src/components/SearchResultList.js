import React, {useState} from "react"
import config from "react-global-configuration"
import {ReactiveList, StateProvider} from "@appbaseio/reactivesearch"
import Grid from "@mui/material/Grid"

import Card from "./Card"
import {OersiConfigContext} from "../helpers/use-context"
import {getParams} from "../helpers/helpers"
import PageControl from "./PageControl"
import {useNavigate, useLocation} from "react-router-dom"

/**
 * Result Component
 * creates a Result box UI component that is used to show the result in base of search,
 * configuration fields are set in src/config/prod.json#resultList
 *
 * @author Edmond Kacaj <edmondikacaj@gmail.com>
 * @props Properties from Parent Component
 */
const SearchResultList = (props) => {
  const location = useLocation()
  const navigate = useNavigate()
  const oersiConfig = React.useContext(OersiConfigContext)
  //declare varibale to get data from Configuration fle prod.json
  const [conf] = useState(config.get("resultList"))
  const [pageSize, setPageSize] = useState(determineInitialPageSize())
  const defaultQuery = function () {
    return {
      track_total_hits: oersiConfig.TRACK_TOTAL_HITS
        ? oersiConfig.TRACK_TOTAL_HITS
        : true,
    }
  }
  return (
    <>
      <ReactiveList
        componentId={conf.component}
        dataField={conf.dataField}
        stream={false}
        pagination={conf.pagination}
        paginationAt={conf.paginationAt}
        pages={conf.pagesShow}
        sortBy={conf.sortBy}
        size={pageSize}
        showLoader={false}
        showEndPage={conf.showEndPage}
        URLParams={conf.URLParams}
        showResultStats={false}
        renderError
        react={{and: conf.and}}
        defaultQuery={defaultQuery}
        sortOptions={conf.sortByDynamic}
        renderNoResults={() => ""}
        renderPagination={({
          pages,
          totalPages,
          currentPage,
          setPage,
          fragmentName,
        }) => {
          return (
            <StateProvider
              componentIds={conf.component}
              includeKeys={["hits"]}
              strict={false}
              render={({searchState}) => (
                <PageControl
                  page={currentPage + 1}
                  total={
                    searchState.results?.hits?.total
                      ? searchState.results?.hits?.total
                      : 0
                  }
                  pageSizeOptions={oersiConfig.RESULT_PAGE_SIZE_OPTIONS}
                  pageSize={pageSize}
                  onChangePage={(page) => {
                    setPage(page - 1)
                  }}
                  onChangePageSize={(size) => {
                    setPageSize(parseInt(size))
                    const params = new URLSearchParams(location.search)
                    params.set("size", size)
                    params.set(conf.component, "1")
                    navigate({
                      pathname: "/",
                      search: "?" + params.toString(),
                    })
                  }}
                />
              )}
            />
          )
        }}
      >
        {({data, error, loading}) => (
          <Grid container direction="row" alignItems="flex-start">
            {data.map((item) => (
              <Grid key={item._id} item xs={12} sm={6} md={4} lg={3} xl={2}>
                <Card {...item} />
              </Grid>
            ))}
          </Grid>
        )}
      </ReactiveList>
    </>
  )
  function determineInitialPageSize() {
    const sizeParam = getParams(location, "size")
    if (
      sizeParam != null &&
      oersiConfig.RESULT_PAGE_SIZE_OPTIONS.indexOf(sizeParam) !== -1
    ) {
      return parseInt(sizeParam)
    } else {
      return oersiConfig.NR_OF_RESULT_PER_PAGE
    }
  }
}

SearchResultList.propTypes = {}

export default SearchResultList
