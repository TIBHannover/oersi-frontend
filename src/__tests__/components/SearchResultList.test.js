import React from "react"
import SearchResultList from "../../components/SearchResultList"
import {render} from "@testing-library/react"
import {SearchIndexFrontendConfigContext} from "../../helpers/use-context"
import {MemoryRouter} from "react-router"
import {getDefaultSearchConfiguration} from "../helpers/test-helpers"

const defaultConfig = {
  GENERAL_CONFIGURATION: {
    NR_OF_RESULT_PER_PAGE: 12,
  },
}

jest.mock("@appbaseio/reactivesearch", () => ({
  ReactiveList: () => <div />,
}))
jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        t: (str) => str,
        changeLanguage: () => new Promise(() => {}),
      },
    }
  },
}))

describe("SearchResultList ==> Test UI  ", () => {
  it("SearchResultList : should render correctly", () => {
    render(
      <MemoryRouter>
        <SearchIndexFrontendConfigContext.Provider
          value={{
            ...defaultConfig.GENERAL_CONFIGURATION,
            searchConfiguration: getDefaultSearchConfiguration(),
          }}
        >
          <SearchResultList />
        </SearchIndexFrontendConfigContext.Provider>
      </MemoryRouter>
    )
  })
})
