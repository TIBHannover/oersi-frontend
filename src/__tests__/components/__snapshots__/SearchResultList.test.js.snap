// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultList ==> Test UI   SearchResultList : should render correctly 1`] = `ShallowWrapper {}`;
